### PR TITLE
Sparse retrieval

### PR DESCRIPTION
**Pull request by adityarg**
This PR contains changes to add an optional feature of sparse-retrieval using Qdrant.
Qdrant recently added the capability to index sparse vectors - [here](https://qdrant.tech/articles/sparse-vectors/)
I am currently using qdrant's [fastembed](https://qdrant.github.io/fastembed/examples/SPLADE_with_FastEmbed/) to get sparse embedding.

### CODE REVIEW

**Comment by Mohannadcse** [comment]
Thanks for the PR @adityarg , can you please add a unit test to validate these new features sparse and dense reponses

**Comment by adityarg** [comment]
Sure @Mohannadcse. Can you guide me on how to add a unit test?
I tested this feature by hosting qdrant container locally. I have the code to create a hybrid collection, index and retrieve chunks for some input query.

**Comment by Mohannadcse** [comment]
Please check these unit tests where we validated Qdrant (both local and cloud) from various perspectives. 
https://github.com/langroid/langroid/blob/main/tests/main/test_vector_stores.py#L149

**Comment by adityarg** [comment]
I've added a new function designed to test hybrid retrieval. This function differs from the previous ones in that it includes two additional variables that enables sparse retrieval.
Unfortunately, due to the issues documented [here](https://discord.com/channels/1131210624479805452/1231878354475552829/1231878354475552829), I couldn't conduct the unit tests successfully.
If you could advise on any necessary modifications on my part, I would appreciate it.
Additionally, please note that There is a dependency of [fastembed](https://github.com/qdrant/fastembed), which is used for getting sparse embedding from the SPLADE model.

**Comment by pchalasani** [comment]
@adityarg I haven't yet got a chance to look at this PR. Will do so later this week.

**Comment by adityarg** [comment]
Sure @pchalasani , I identified a issue with backward compatibility with my PR. I pinged qdrant regarding it and they provided a solution for it, I will push the changes by tomorrow.
You can check the conversation with qdrant [here](https://discord.com/channels/907569970500743200/1234403379992723487/1234403379992723487)

**Comment by adityarg** [comment]
I have pushed 2 changes:
1. Now using `huggingface` to use SPLADE model instead of `fastembed`, since `fastembed` does not the latest splade-v3 models
2. I have replaced dense vectors names from "text-dense" to "" (empty string). This makes it compatible to the collection which have only dense by default.

**Comment by pchalasani** [comment]
@adityarg I recently changed `qdrantdb.py`, to catch up with the fact that `recreate_collection` is no longer recommended. Your PR is based on the version prior to my change, which is causing a conflict. Rather than attempt to patch the conflict (which may be messy), could you please pull latest `qdrantdb.py` from main and build your PR on top of that? Thanks.

File: langroid/vector_store/qdrantdb.py [hunk 1]
```diff
@@ -220,6 +246,28 @@ def create_collection(self, collection_name: str, replace: bool = False) -> None
             logger.setLevel(logging.INFO)
             logger.info(collection_info)
             logger.setLevel(level)
+    
+    def get_sparse_embeddings(self, inputs: List[str]) -> List[SparseVector]:
+        if not self.config.use_sparse_embeddings: return []
+        import torch
+        tokens = self.sparse_tokenizer(inputs, return_tensors='pt', truncation=True, padding=True)
+        output = self.sparse_model(**tokens)
+        vectors = torch.max(
```
Comment (pchalasani) [inline langroid/vector_store/qdrantdb.py:255]: formatting is missing indents. Make sure to run `make lint` to fix style issues.

**Comment by adityarg** [comment]
@pchalasani I have built the PR from the latest code and fixed styling issues in this PR - https://github.com/langroid/langroid/pull/462

**Pull request closed without merging**